R/zchunk_L2321.cement.R

```diff
@@ -73,243 +70,360 @@ module_energy_L2321.cement_DISABLED <- function(command, ...) {
+
+    # Carbon capture rates from technologies with CCS
+    # L2321.GlobalTechCapture_cement: CO2 capture fractions from global cement production technologies with CCS
+    # No need to consider historical periods or intermittent technologies here
+    A321.globaltech_co2capture %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      complete(nesting(supplysector, subsector, technology), year = c(year, FUTURE_YEARS)) %>%
+      arrange(supplysector, subsector, technology, year) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(remove.fraction = approx_fun(year, value, rule = 1),
+             remove.fraction = round(remove.fraction, energy.DIGITS_REMOVE.FRACTION)) %>%
+      ungroup %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      rename(sector.name = supplysector,
+             subsector.name = subsector) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["GlobalTechYr"]], "remove.fraction"))) %>%
+      mutate(storage.market = energy.CO2.STORAGE.MARKET) ->
+      L2321.GlobalTechCapture_cement
+
+    # L2321.GlobalTechCost_cement: Non-energy costs of global cement manufacturing technologies
+    A321.globaltech_cost %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      complete(nesting(supplysector, subsector, technology, minicam.non.energy.input), year = c(year, BASE_YEARS, FUTURE_YEARS)) %>%
+      arrange(supplysector, subsector, technology, minicam.non.energy.input, year) %>%
+      group_by(supplysector, subsector, technology, minicam.non.energy.input) %>%
+      mutate(input.cost = approx_fun(year, value, rule = 1),
+             input.cost = round(input.cost, energy.DIGITS_COST)) %>%
+      ungroup %>%
+      filter(year %in% c(BASE_YEARS, FUTURE_YEARS)) %>%
+      rename(sector.name = supplysector,
+             subsector.name = subsector) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["GlobalTechCost"]])) ->
+      L2321.GlobalTechCost_cement # intermediate tibble
+
+    # Note: adjusting non-energy costs of technologies with CCS to include CO2 capture costs
+    # Note: The additional CCS-related non-energy costs are not included in the global technology assessment. Calculate here.
+    # First calculate the additional CCS costs per unit of carbon produced in 1975$
+    cement_CCS_cost_total_1975USDtC <- energy.CEMENT_CCS_COST_2000USDTCO2 * CONV_2000_1975_USD * CONV_C_CO2
+    CO2_storage_cost_1975USDtC <- energy.CO2_STORAGE_COST_1990_USDTC * CONV_1990_1975_USD
+    cement_CCS_cost_1975USDtC <- cement_CCS_cost_total_1975USDtC - CO2_storage_cost_1975USDtC
 
+    # Next calculate the quantity of CO2 produced per unit of cement produced (in kgC per kg cement)
+    cement_CO2_capture_frac <- mean( L2321.GlobalTechCapture_cement[["remove.fraction"]] )
+    CO2_IO_kgCkgcement <- mean(unlist(select(filter(L2321.GlobalTechCoef_cement, minicam.energy.input == "limestone"), coefficient))) *
```

Comment:
This should be clarified; also rather that `unlist` we've been using `.[["coefficient"]]` at end of a pipeline.

```R
L2321.GlobalTechCoef_cement %>%
  filter() %>%
  select() %>%
  .[["coefficient"]] %>%
  mean ->
  CO2_IO_kgCkgcement
```